crates/biome_js_analyze/src/lint/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,353 @@
+        let attributes_ref = attributes.as_ref().unwrap_or(&default_attributes);
+
+        let attributes_option = if !attributes_ref.is_empty() {
+            Some(attributes_ref.clone())
+        } else {
+            None
+        };
+
+        if has_truthy_ariahidden_attr(node) {
+            return None;
+        }
+
+        let is_valid_element = match elm {
+            "section" => ["aria-label", "aria-labelledby"].iter().any(|&attr_name| {
+                node.find_attribute_by_name(attr_name)
+                    .map_or(false, |attr| {
+                        attr.as_static_value()
+                            .map_or(false, |val| !val.text().is_empty())
+                    })
+            }),
+            "a" => node.find_attribute_by_name("href").map_or(false, |attr| {
+                attr.as_static_value()
+                    .map_or(false, |val| !val.text().is_empty())
+            }),
+            _ => {
+                (!aria_roles.is_not_interactive_element(
+                    element_name.text_trimmed(),
+                    attributes_option.clone(),
+                ) && !is_invalid_element(elm))
+                    || is_valid_element(elm)
+            }
+        };
+
+        if is_valid_element {
+            return None;
+        }
+
+        if attributes_option.is_some() {
+            match node.find_attribute_by_name("role") {
+                Some(attr) => {
+                    let role_value = attr.as_static_value()?;
+                    let role_text = role_value.text();
+
+                    if has_valid_role(role_text) || aria_roles.is_role_interactive(role_text) {
+                        return None;
+                    }
+                }
+                None => {
+                    return Some(());
+                }
+            };
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {{"Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element"}},
+        ))
+    }
+}
+
+/// Check if any interactive handler is present in the attributes.
+fn is_interactive_handler_present(attributes: &FxHashMap<String, Vec<String>>) -> bool {
+    let categories_to_check = vec!["focus", "keyboard", "mouse"];
+    for category in categories_to_check {
+        if let Some(handlers) = INTERACTIVE_HANDLER_LIST.get(category) {
+            for handler in handlers {
+                if let Some(values) = attributes.get(*handler) {
+                    if values.iter().any(|value| value != "null") {
+                        return true;
+                    }
+                }
+            }
+        }
+    }
+    false
+}
```

Comment:
This function used once, so I think it's enough to write directly in the `run`. We can use `Option` to simplify this function in the `run`
And, `categories_to_check` should be initialized with `const` outside the function.

Reply:
@unvalley 
fixed in this commit 22819330bb9b1966dab553c89e5d0dbb0b093d2e